test/sequential/test-child-process-execsync.js

```diff
@@ -12,6 +12,13 @@ var start = Date.now();
 var err;
 var caught = false;
 
+// Verify that stderr is not accessed when a bad shell is used
+(function() {
+  assert.throws(function() {
+    execSync('exit -1', {shell: 'no_shell'});
+  }, /spawnSync no_shell ENOENT/);
+})();
+
```

Comment:
Yes, there is a similar test down below, but I put this test the first one, because it has to be ran before other calls to execSync.
For example, if the below test, with timeout, is ran before the one added in this patch, it could throw another exception, masking the real one.